### ISSUE TITLE
Pathname completion and commands taking pathnames as arguments

### ISSUE THREAD

**Issue by deterministic-arts** [McCLIM/McCLIM#876]
The accept implementation for `pathname` values on input streams seems to have some problems recently (I am sure it worked differently at some point in the past.) To reproduce, fire up a `clim-listener` and 

    (define-command (com-dummy :command-table 'clim-listener::listener :name t)
        ((arg 'pathname))
      (print arg))

When trying to execute the command in the REPL (via `,` prefix), I can enter a file name, but I have to hit the activation key (return) twice for the command for anything to happen.

    (clim:define-command (com-dummy :command-table clim-listener::listener :name t)
        ((arg 'pathname)
         (arg2 'string))
      (print (list arg arg2)))

In this version, I have to enter the file name, and then hit the activation key to get the inline prompt for the second argument.

I am not sure whether these effects are deliberate or not. They make pathname input very different from how things are handled for (most) other presentation types. In particular, they make entering command arguments "in-line" quite inconvenient as soon as pathnames are involved.

**Comment by scymtym** [comment]
I could reproduce the behavior your are describing, but only if the current value of the `pathname` argument end with `/`, i.e.

```
,Dummy /tmp RET -> Command executes
,Dummy /tmp/ RET RET -> Command executes
```

Is this consistent with your observations?

**Comment by deterministic-arts** [comment]
Not quite. For me, this shows up also, when I just type a simple name like

    Command> ,Dummy (pathname) foo.txt

(at this point, I am required to hit RET twice to make the command proceed). The same for the second example:

    Command> ,Dummy (pathname) foo.txt (string) bar

I have to hit RET after `foo.txt` instead of the more customary SPACE here to get prompted for the next argument. 

On an unrelated note, the command parameter dialog (accepting values thing), that usually happens, if you hit the activation gesture right after the command (and before the first parameter), i.e., 

    Command> ,Dummy

seems to be broken in the listener. I now get a `NO-APPLICABLE-METHOD-ERROR` complaining about a missing method on `CLIM:ENCAPSULATING-STREAM-STREAM`. The listener window shows the initial 

    You are being prompted for arguments to "Dummy"

and then breaks into the debugger. This is probably unrelated, though.

**Comment by deterministic-arts** [comment]
P.S.: Note, that I do not try any completion for the files in question; I simply type the file name directly. The phenomenon exists regardless of whether there is a matching file or not.

**Comment by scymtym** [comment]
> Not quite.

Right. I see now that the behavior is more complicated. What I have observed so far:

```
/non-existing-directory             -> RET RET
/non-existing-directory/            -> RET RET

/existing-directory                 -> RET
/existing-directory/                -> RET RET

relative-name-of-non-existing-file  -> RET RET
/absolute-name-of-non-existing-file -> RET RET

relative-name-of-existing-file      -> RET RET
/absolute-name-of-existing-file     -> RET
```

> On an unrelated note, the command parameter dialog (accepting values thing), that usually happens, if you hit the activation gesture right after the command (and before the first parameter), i.e.,
>
>     Command> ,Dummy
>
> seems to be broken in the listener. I now get a `NO-APPLICABLE-METHOD-ERROR` complaining about a missing method on `CLIM:ENCAPSULATING-STREAM-STREAM`. The listener window shows the initial
>
>     You are being prompted for arguments to "Dummy"
>
> and then breaks into the debugger. This is probably unrelated, though.

Does this happen with the current master version of McCLIM? I believe we fixed this very recently.

**Comment by deterministic-arts** [comment]
I did not test this against the current master. The version I use came via the last quicklisp dist update.
